js/src/launchAsync.ts

```diff
@@ -63,6 +74,273 @@ let launchResponseResolve: LaunchResponseResolve;
 let launchResponseReject: LaunchReject;
 let launchResponseError: Error;
 
+// function _checkIfLoading(): Promise<LaunchResponse> | null {
+//     if (isLoading) {
+//         return Promise.reject('Immersive Reader is already launching');
+//     }
+//     return null;
+// }
+
+// Added globals to persist between functions
+let timeoutId: number | null = null;
+let parent: Node = null;
+let iframeContainer: HTMLDivElement = null;
+let iframe: HTMLIFrameElement = null;
+let noscroll: HTMLStyleElement = null;
+let messageHandler: (e: any) => void = null;
+
+enum FuncType { launchAsync, launchWithoutContentAsync }
+
+function _initializeOptions(options?: Options): Options {
+
+    const { allowFullscreen, cookiePolicy, hideExitButton, onExit, parent, timeout, uiZIndex, useWebview } = options || {};
+
+    // FIXME: will this options set work?
```

Comment:
@dylankil  FYI the reason the comment was here was because of the order of operations -> in this case the difference from existing code is that i default with the spread of ...options and then i set the defaults.